src/components/Gallery.tsx

```diff
@@ -23,9 +27,20 @@ export default class Gallery extends Component<GalleryProps, GalleryState> {
     this.state = {
       fullUrls: [],
       renderPlaceholder: true,
+      selectedSource: '',
     };
   }
 
+  async componentDidMount() {
+    this.renderImagesOrPlaceholder();
+  }
+
+  async componentDidUpdate(prevProps: GalleryProps) {
+    if (this.props.selectedSource.id !== prevProps.selectedSource.id) {
+      this.renderImagesOrPlaceholder();
+    }
+  }
+
```

Comment:
Moved React component lifecycle methods to top of file to match community convention.